momepy/tests/test_elements.py

```diff
@@ -154,8 +155,16 @@ def test_get_node_id(self):
         self.df_buildings["nID"] = mm.get_network_id(
             self.df_buildings, self.df_streets, "nID"
         )
-        ids = mm.get_node_id(self.df_buildings, nodes, edges, "nodeID", "nID")
-        assert not ids.isna().any()
+        ids1 = mm.get_node_id(self.df_buildings, nodes, edges, "nodeID", "nID")
+        assert not ids1.isna().any()
+
+        # test for NaNs within `object` nIDs column
+        _df_buildings = self.df_buildings.copy()
+        _df_buildings.loc[[0, 1], "nID"] = pd.NA
```

Comment:
```suggestion
        _df_buildings["nID"] = _df_buildings["nID"].astype(str)
        _df_buildings.loc[[0, 1], "nID"] = pd.NA
```

Might be worth actually testing for the string ID?

Reply:
Sure. Will push up a new version.